packages/itmat-ui-react/src/components/lxd/lxd.instance.terminal.tsx

```diff
@@ -0,0 +1,279 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { XTerm } from 'xterm-for-react';
+import { FitAddon } from 'xterm-addon-fit';
+import 'xterm/css/xterm.css';
+import axios from 'axios';
+import { updateMaxHeight } from './util/updateMaxHeight';
+import './lxd.module.css';
+
+export const LXDCommandExecutor: React.FC<{ instanceName: string }> = ({ instanceName }) => {
+    // const [dataWs, setDataWs] = useState<WebSocket | null>(null);
+    // const [controlWs, setControlWs] = useState<WebSocket | null>(null);
+
+    const dataWs = useRef<WebSocket | null>(null);
+    const controlWs = useRef<WebSocket | null>(null);
+
+    const xtermRef = useRef<XTerm>(null);
+    const [terminalReady, setTerminalReady] = useState(false);
+    // const fitAddon = useRef(new FitAddon()).current;
+    const fitAddonRef = useRef(new FitAddon());
+    const textEncoder = new TextEncoder();
+
+
+    // Create a ref for terminal initialization flag
+    const isTerminalInitialized = useRef(false);
+
+
+    const backendEndpoint = 'http://localhost:3333/api/lxd';
+    // const backendEndpoint = 'https://192.168.64.4:8443/1.0';
+
+
+    // Initialize WebSocket connection
+    const initWebSocket = async () => {
+        try {
+
+            const execPayload = {
+                'command': ['bash'],
+                'environment': {
+                    TERM: 'xterm-256color',
+                    HOME: '/root'
+                },
+                'user': 0, // Use provided user or default to 0
+                'group': 0, // Use provided group or default to 0
+                'wait-for-websocket': true,
+                'interactive': true
+            };
+
+            const response = await axios.post(`${backendEndpoint}/instances/${instanceName}/exec`, execPayload
+            );
+            const { fds, operationId } = response.data;
+            console.log('WebSocket request response:', response.data);
+            // console.log('Frontend code for response', response);
+            if (fds && operationId) {
+
+                const lxdServerWsBase = 'ws://localhost:3333/ae_wslxd';
```

Comment:
What is this endpoint supposed to be ?

Reply:
the terminal module has been removed and replaced by the text console.